File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.py
class Solution:
    def expand(self, s: str, l: int, r: int):
        while l >= 0 and r < len(s) and s[l] == s[r]:
            l -= 1
            r += 1
        return l + 1, r - 1
    def longestPalindrome(self, s: str) -> str:
        maxLength = 1
        start = 0
        end = 0
        for i in range(len(s)):
            l, r = self.expand(s, i, i + 1)
            if r - l + 1 > maxLength:
                start = l
                end = r
                maxLength = r - l + 1
            l, r = self.expand(s, i, i)
            if r - l + 1 > maxLength:
                start = l
                end = r 
                maxLength = r - l + 1
        return s[start:end + 1]

        

File: 0005-longest-palindromic-substring/0005-longest-palindromic-substring.cpp
class Solution {
public:
    string longestPalindrome(string s) {
        int maxLength = 1;
        int start = 0;
        for (int i = 0; i < s.length(); i++) {
            int l = i;
            int r = i + 1;
            while (l >= 0 && r < s.length()) {
                if (s[l] == s[r]) {
                    if (r - l + 1 > maxLength) {
                        maxLength = r - l + 1;
                        start = l;
                    }
                    l--;
                    r++;
                } else {
                    break;
                }
            }

            l = i - 1;
            r = i + 1;
            while (l >= 0 && r < s.length()) {
                if (s[l] == s[r]) {
                    if (r - l + 1 > maxLength) {
                        maxLength = r - l + 1;
                        start = l;
                    }
                    l--;
                    r++;
                } else {
                    break;
                }
            }
        }
        return s.substr(start, maxLength);
    }
};